src/psyclone/psyir/symbols/interfaces.py

```diff
@@ -110,14 +110,31 @@ class ImportInterface(SymbolInterface):
         from which the symbol is imported.
     :type container_symbol: \
         :py:class:`psyclone.psyir.symbols.ContainerSymbol`
+    :param Optional[str] orig_name: the name of the symbol in the \
+        external container before it is renamed, or None (the default) if \
+        it is not renamed.
 
+    :raises TypeError: if the orig_name argument is an unexpected type.
 
     '''
-    def __init__(self, container_symbol):
+    def __init__(self, container_symbol, orig_name=None):
         super().__init__()
+        if not isinstance(orig_name, (str, type(None))):
+            raise TypeError(
+                f"ImportInterface orig_name parameter must be of type "
+                f"str or None, but found '{type(orig_name).__name__}'.")
+        self._orig_name = orig_name
         # Use error-checking setter
         self.container_symbol = container_symbol
 
+    @property
+    def orig_name(self):
+        '''
+        :returns: the symbol's original name if it has been renamed.
```

Comment:
s/has been renamed/is renamed upon import or None otherwise/

Reply:
Modified.